lib/transports/websocket.js

```diff
@@ -10,12 +10,12 @@ var yeast = require('yeast');
 var debug = require('debug')('engine.io-client:websocket');
 
 /**
- * `ws` exposes a WebSocket-compatible interface in
- * Node, or the `WebSocket` or `MozWebSocket` globals
- * in the browser.
+ * Get either the `WebSocket` or `MozWebSocket` globals
+ * in the browser or the WebSocket-compatible interface
+ * exposed by `ws` for Node environment.
  */
 
-var WebSocket = require('ws');
+var WebSocket = global.WebSocket || global.MozWebSocket || require('ws');
```

Comment:
To comply with https://github.com/websockets/ws/commit/e54d45fbab3b19c2940e9057ce1e7b8f105873e0